pkg/upgrade/upgrade.go

```diff
@@ -27,9 +26,24 @@ type NodeVersion struct {
 	DriverToolkit  registry.DriverToolkitEntry `json:"driverToolkit"`
 }
 
-func ClusterInfo() (map[string]NodeVersion, error) {
+type ClusterInfo interface {
+	GetClusterInfo() (map[string]NodeVersion, error)
+}
+
+func NewClusterInfo() *clusterInfo {
```

Comment:
I think it is better to return the intefrace ClusterInfo, then the inner struct clusterInfo

Reply:
You are absolutely right. This doesn't make much sense right now